lib/maze/server.rb

```diff
@@ -241,15 +241,22 @@ def stop
       end
 
       def reset!
-        Maze::Server.errors.clear
-        Maze::Server.sessions.clear
-        Maze::Server.builds.clear
-        Maze::Server.uploads.clear
-        Maze::Server.sourcemaps.clear
-        Maze::Server.traces.clear
-        Maze::Server.logs.clear
-        Maze::Server.invalid_requests.clear
-        @generators&.values&.each { |generator| generator.close }
+        # Default to HTTP 200 return status after each scenario
+        set_status_code_generator(Maze::Generator.new [Maze::Server::DEFAULT_STATUS_CODE].cycle)
+
+        # Similarly for the response delay
+        @response_delay_ms = 0
+        @reset_response_delay = false
+        errors.clear
+        sessions.clear
+        builds.clear
+        uploads.clear
+        sourcemaps.clear
+        traces.clear
+        logs.clear
+        invalid_requests.clear
+        @generators&.values.each(&:close)
+        @generators = nil
```

Comment:
Need to close the generators before overwriting them

```suggestion
        @generators&.values.each(&:close)
        @response_delay_ms = 0
        @reset_response_delay = false
        errors.clear
        sessions.clear
        builds.clear
        uploads.clear
        sourcemaps.clear
        traces.clear
        logs.clear
        invalid_requests.clear

        # Default to HTTP 200 return status after each scenario
        set_status_code_generator(Maze::Generator.new [Maze::Server::DEFAULT_STATUS_CODE].cycle)
```

Reply:
Oh actually, if we're overwriting them then we'll already handle closing them, so this can be:

```suggestion
        # Default to HTTP 200 return status after each scenario
        set_status_code_generator(Maze::Generator.new [Maze::Server::DEFAULT_STATUS_CODE].cycle)

        # Similarly for the response delay
        @response_delay_ms = 0
        @reset_response_delay = false
        errors.clear
        sessions.clear
        builds.clear
        uploads.clear
        sourcemaps.clear
        traces.clear
        logs.clear
        invalid_requests.clear
```

Reply:
Oops, of course - mindless cut/paste of other reset code.

Reply:
2nd commit now pushed for the follow-up comment.